pkg/tfbridge/schema.go

```diff
@@ -798,3 +794,49 @@ func CleanTerraformSchema(tfs map[string]*schema.Schema) map[string]*schema.Sche
 	}
 	return cleaned
 }
+
+// CoerceTerraformStringOutput coerces a field with the given Terraform schema to a PropertyValue. Terraform often
+// produces Go values of type String for resources whose schema indicates that the type is not String - usually Int,
+// Bool, or Float. In this case, CoerceTerraformStringOutput inspects the schema and, if the schema is present, coerces
+// the string value to the requested type and returns a PropertyValue for it.
+//
+// If the schema is not present (i.e. this is a Pulumi-specific field), there is an error coercing the string value,
+// or the schema indicates the field is not of type Int, Bool, or Float, the string is returned verbatim.
+func CoerceTerraformStringOutput(sch *schema.Schema, stringValue string) resource.PropertyValue {
```

Comment:
How about something like

```suggestion
func CoerceTerraformString(to schema.ValueType, s string) interface{} {
    switch sch.Type {
    case schema.TypeInt:
        ...
    case schema.TypeBool:
        ...
    case schema.TypeFloat:
        ...
    default:
        return s
    }
}
```

This will make it easier to consume from `tf2pulumi` (or other libraries that don't deal in `resource.PropertyValue`).

Reply:
sounds good, will do.